src/stdpar/plugin-SiPixelRawToDigi/SiPixelDigisSoAFromCUDA.cc

```diff
@@ -50,21 +42,12 @@ void SiPixelDigisSoAFromCUDA::acquire(const edm::Event& iEvent,
   const auto& gpuDigis = ctx.get(iEvent, digiGetToken_);
 
   nDigis_ = gpuDigis.nDigis();
-#ifdef CUDAUVM_DISABLE_MANAGED_CLUSTERING
-  pdigi_ = gpuDigis.pdigiToHostAsync(ctx.stream());
-  rawIdArr_ = gpuDigis.rawIdArrToHostAsync(ctx.stream());
-  adc_ = gpuDigis.adcToHostAsync(ctx.stream());
-  clus_ = gpuDigis.clusToHostAsync(ctx.stream());
-#else
-  gpuDigis.pdigiPrefetchAsync(cudaCpuDeviceId, ctx.stream());
-  gpuDigis.rawIdArrPrefetchAsync(cudaCpuDeviceId, ctx.stream());
-  gpuDigis.adcPrefetchAsync(cudaCpuDeviceId, ctx.stream());
-  gpuDigis.clusPrefetchAsync(cudaCpuDeviceId, ctx.stream());
+  // This memory is already owned by a unique pointer, therefore class members cannot be unique_ptr themselves
+  // *this must have a shorter lifetime than gpuDigis object retrieved from the context or we risk using dangling pointers
```

Comment:
Just to clarify, the objects retrieved from `edm::Event` are alive until everything has been processed for a given event. In absence of asynchronous execution the use of `edm::EDProducerExternalWork` and the split to `acquire()` and `produce()` functions is not necessary (i.e. all this could be simplified in some subsequent PR).

Reply:
In that case, the class ```edm::EDProducerExternalWork``` would no longer be necessary since its purpose was to provide ```acquire()```?  Parallel algorithms seem to be a similar abstraction to what ```edm::EDProducerExternalWork``` does; offloading the work "somewhere" to parallelize it.

Reply:
> In that case, the class `edm::EDProducerExternalWork` would no longer be necessary since its purpose was to provide `acquire()`?

Correct (and I'd leave the cleanup in a later PR).

> Parallel algorithms seem to be a similar abstraction to what `edm::EDProducerExternalWork` does; offloading the work "somewhere" to parallelize it.

I'd say the abstractions are not quite the same. The part of "offload work somewhere" is indeed similar. But the key point of `edm::EDProducerExternalWork` is that while the offloaded work is being run (i.e. the work has been launched in `acquire()`) the framework (really TBB) can run other CPU tasks on the same CPU worker thread that run `acquire()`. The `produce()` is called after the offloaded work has finished, but not necessary by the same CPU worker thread.

Reply:
Yes we lose the asynchronous offloading of work until the c++ standard library has better async support, most likely with c++26 and senders / receivers.